lib/importer/index.js

```diff
@@ -240,34 +255,38 @@ exports.Initialise = (config, router, prototypeKit) => {
     let maxCol = sheets_lib.GetTotalColumns(session);
 
     // Find the selected range, or default to the first row (0,0)->(0,maxCol)
-    // We should probably find a better way of handling no selection, and we still
-    // need to decide how users can decide between auto-choose header and user-chosen
-    // headers.
-    let tlRow = getIntOrDefault(request.body['importer:selection:TLRow']);
-    let tlCol = getIntOrDefault(request.body['importer:selection:TLCol']);
-    let brRow = getIntOrDefault(request.body['importer:selection:BRRow']);
-    let brCol = getIntOrDefault(request.body['importer:selection:BRCol'], maxCol);
-
-    // Normalise the selection so that start row <= end row and start col <= end col
-    if (tlRow > brRow) {
-      [tlRow, brRow] = [brRow, tlRow];
+    session.headerRange = getSelectionFromRequest(request, session, optional=false);
+
+    // Ensure the session is persisted. Currently in session, eventually another way
+    request.session.data[IMPORTER_SESSION_KEY] = session;
+    redirectOnwards(request, response);
+  });
+
+
+  //--------------------------------------------------------------------
+  // Allows the user to optionally select a footer row should the data
+  // contain one
+  //--------------------------------------------------------------------
+  router.post(IMPORTER_ROUTE_MAP.get("importerSelectFooterPath"), (request, response) => {
+    let session = request.session.data[IMPORTER_SESSION_KEY];
+    if (!session) {
+      response.status(404);
+      return;
     }
-    if (tlCol > brCol) {
-      [tlCol, brCol] = [brCol, tlCol];
+
+    let range = getSelectionFromRequest(request, session, optional=true);
+    if (!range) {
+      redirectOnwards(request, response);
+      return;
     }
 
-    session.headerRange = {
-      sheet: session.sheet,
-      start: {row: parseInt(tlRow), column: parseInt(tlCol)},
-      end: {row: parseInt(brRow), column: parseInt(brCol)},
-    };
+    session.footerRange = range;
```

Comment:
This cannot just be the range we were passed in, it needs to be indexed based on the rows that were shown, which were almost certainly not - indexed.  We need to re-calculate which rows were sent in the previed (via GetTrailingRows) and then increment the indices by that amount. 

Currently we calculate the range (and how many rows to fetch with)

```
    const sheetDimensions = backend.SessionGetInputDimensions(session.backendSid)
          .sheetDimensions.get(session.sheet);

    // We have to calculate the range containing the data based on what we believe
    // the header range to be.
    const hRange = session.headerRange;

    const end = sheetDimensions.rows -1;
    const rowRange = {
        sheet: session.sheet,
        start: { row: Math.max(end-count, hRange.start.row +1), column: hRange.start.column},
        end: { row: end, column: hRange.end.column}
    }
```

So we might want to move this out of GetTrailingRows into a healper function.